backend/src/director.py

```diff
@@ -14,17 +14,13 @@
 
 def question(question):
     logging.debug("Received utterance: {question}")
-
     intent = intent_agent.invoke(question)
-    logging.debug(f"Intent determined: {intent}")
-
-    if determine_intention(question) == "TRUE":
-        task_dict = create_tasks(question, agents_details)
-        final_answer = solve_all_tasks(task_dict)
-        return final_answer
-
-    logging.info("Passing utterance straight to call_model function")
-    return call_model(director_prompt, user_prompt=question)
+    intent_json = json.loads(intent)
+    logging.info(f"Intent determined: {intent}")
+    final_scratchpad = solve_all_tasks(intent_json)
```

Comment:
I wonder whether `solve_all_tasks` should be renamed to `solve_all` so that it is agnostic of tasks vs questions